src/dynamics/video_player/player/states.js

```diff
@@ -440,7 +440,7 @@ Scoped.define("module:VideoPlayer.Dynamics.PlayerStates.Outstream", [
         },
 
         _nextState: function() {
```

Comment:
Hi, @sambua! I hope you are doing well. I looked at the new code for loading ads and from what I understood it's doing the following:

```
if (is floating or user interacted with the player) {
    // load ads
} else {
   // wait until player is in view before loading ads
}
```

There are some cases where the player might be in view and it's not floating, such as when it's fullscreened or displayed inline. Do you think it might be more comprehensive to check if `view_type` is not equal to "out_of_view" instead?

Additionally, I'm curious about the requirement for user interaction before ad loading. Can you explain to me the reasoning behind it? I'm wondering if there's a specific scenario or user experience consideration that I might be missing.

Looking forward to your thoughts on this. 🙂

Reply:
Hi @ntshcalleia , scrollIntoView was added, as before I remember there was trouble when ads start to play, it made to scroll to the video player content container. And it made some trouble on UX way, so why I've added it. 

And there could be situation when floating only set, when there are no video content block and video starts on floating directly so it should start. 

I've recently added start to play when user already has interaction because when user press to replay the ad on outtstream, it mean user had interaction. And if player on fullscreen it means user already has interaction and ads should start before player in the scroll view.

But anyway if you see that it's causing any bug feel free to make required changes. I've just explain above what was a reason to adding this statement.